channels/sig_analog.c

```diff
@@ -1766,6 +1766,7 @@ static void *__analog_ss_thread(void *data)
 
 	if (!chan) {
 		/* What happened to the channel? */
+		ast_log(LOG_WARNING, "No Asterisk channel provided for DAHDI channel %d?\n", p->channel);
```

Comment:
This is not something the user can fix so it should be a debug message

Reply:
I disagree... this means that something is wrong, and I feel it would be best to make some noise about this, since this should never happen. If there had been a warning here in the first place, this issue would have been resolved much more quickly.

I could make it a `LOG_ERROR` instead of a `LOG_WARNING`, since it really is an error of something wrong and not a warning of user misconfig.

`[2023-11-23 21:35:30.623] WARNING[1131310]: chan_dahdi.c:12011 do_monitor: Whoa....  I'm owned by DAHDI/12-1 but found (29) in read...` isn't something the user can fix either, but that's what clued me into this problem in the first place. Making it a debug message would just hide something that needs to be dealt with.

Reply:
Original comment still applies

Reply:
And I still disagree. Most `LOG_ERROR`'s in Asterisk are signs of something wrong with Asterisk itself, not an indication of something the user did or could do something about. This is a case where something is wrong with Asterisk itself. This is being consistent with existing logging practices, I see no reason to deviate from that.

If someone else still wants it to be a debug, I'll change it, but I don't think that's the right thing to do.

Reply:
It’s not going in unless it’s a debug.

Reply:
I agree with Sean, as well in the past log messages were not really thought about, reviewed, or considered. For log messages that are presented to users they should be informative to them and provide guidance if possible. This log message as it exists now is logging out an implementation detail aspect for developers and would be confusing for users.